### PR TITLE
Fixes TV > Tpl Access sorting bug

### PR DESCRIPTION
**Pull request by davidpede**
### What does it do?
Fixes v old bug by setting remoteSort to false in the 'TV>Template Access' grid

### Why is it needed?
When sorting by the Access column the processor returns no results and xpdo logs an unknown column error. This because the access value is set manually in the processor after the db query as it is not stored in db table: https://github.com/modxcms/revolution/blob/5c59d5bc4f2ffd77454d2a42a1d1046ba2f7d808/core/model/modx/processors/element/tv/template/getlist.class.php#L101

### Related issue(s)/PR(s)
Fixes: #12319, #7896

### CODE REVIEW

**Comment by Jako** [comment]
Instead of disabling remote sort, the access column should not be sortable in https://github.com/davidpede/revolution/blob/ed8038bfdc75d840d2d624a430a665b725748b88/manager/assets/modext/widgets/element/modx.grid.tv.template.js#L15

**Comment by Jako** [comment]
But before you disable sortable of the access column: How expensive would it be to create a sortable `template_access` column in the query and reference it in the grid?

**Comment by davidpede** [comment]
@Jako How does last commit look for a quick solution? It seem to work as `tmplvarid` is either the current tv id or null for that query

File: core/model/modx/processors/element/tv/template/getlist.class.php [hunk 1]
```diff
@@ -27,6 +27,11 @@ public function initialize() {
             'dir' => 'ASC',
             'tv' => false,
         ));
+        if ($this->getProperty('sort') == 'access') {
+          $this->setProperties(array(
```
Comment (Jako) [inline core/model/modx/processors/element/tv/template/getlist.class.php:31]: $this->setProperty('sort', 'tmplvarid');

**Review by Jako — COMMENTED** [review]
Should be almost suitable. But since `tmplvarid` could be an integer, the sorting depends not only on true/false but on the integer value.

You could add a select query with 
```
$c->select(array('access' => 'IF(`TemplateVarTemplates.tmplvarid`,true,false)'));
```
in the getData method, but I don't know wether this is only MySQL syntax.

**Comment by Mark-H** [comment]
Sounds this is still pending improvements, @davidpede?

**Comment by davidpede** [comment]
@Mark-H Yes leave this for now, I'll look at jako's suggestions asap

**Comment by gpsietzema** [comment]
@davidpede MODX 2.7 is coming soon, do you see any chance to change this PR on short term?

**Comment by davidpede** [comment]
Can look at this next week if soon enough? @gpsietzema

**Comment by gpsietzema** [comment]
That would work if it is early next week @davidpede - Jason is planning on merging everything in this and next week. Believe me, it's a sh*tload of work ;-)

**Comment by davidpede** [comment]
@gpsietzema @Jako I completely screwed this branch so have re-opened here #13893

**Pull request closed without merging**